### PR TITLE
refactor : remove moment in favor of dayjs

### PR DESCRIPTION
**Pull request by shftrg**
Resolve opencollective/opencollective#2446

# Description

# Screenshots

### CODE REVIEW

**Comment by darkmtr** [comment]
@Betree i have made the initial changes, I'd like your input. As for the poly fill for intl.DateTimeFormate [this](https://github.com/formatjs/date-time-format-timezone) is recommended by dayjs.

File: components/expenses/TransactionsExportPopoverAndButton.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ class ExportForm extends React.Component {
     super(props);
 
     // Calculate the start: first day of previous month & end date: today.
-    const oneMonthAgo = moment().subtract(1, 'months')._d;
+    const oneMonthAgo = dayjs().subtract(1, 'month').$d;
```
Comment (Betree) [inline components/expenses/TransactionsExportPopoverAndButton.js:80]: `toDate` would be clearer

```suggestion
    const oneMonthAgo = dayjs().subtract(1, 'month').toDate();
```

**Review by Betree — COMMENTED** [review]
Haven't tested, but overall the change look good to me. I like how the usage with dayjs is similar to `moment`, that will make the transition easier.

@znarf wdyt?

**Comment by darkmtr** [comment]
@Betree will make the requested changes. what about the polyfill for Intl.DateTimeFormat?

**Comment by Betree** [comment]
> @Betree will make the requested changes. what about the polyfill for `Intl.DateTimeFormat`?

That's ok to me

**Review by Betree — DISMISSED** [review]
Editing the time for an event seems broken.
Also the default values are different from what we have on staging.

Peek 2020-10-02 10-44

**Comment by darkmtr** [comment]
@Betree i have fixed the issue now. Please let me know if you find more discrepancies, i couldn't test it properly as i'm busy with school.

**Comment by Betree** [comment]
@darkmtr We can help reviewing, but testing is part of the development task when you do a bounty. You need to test your changes and make sure that they work, otherwise it's more work for us to test everything and to provide feedback than it is to just do the task directly. Please test the paths that you've updated when you'll have more time.

**Comment by Betree** [comment]
Hello @darkmtr, any news on this one?

**Comment by snitin315** [comment]
Ready for review 🙌🏻  

/cc @Betree

**Comment by Betree** [comment]
The time picker for events still acts strangely, jumping multiple numbers at once. 

Peek 2020-11-02 14-34

**Comment by snitin315** [comment]
It's working fine in my local env. I am using Linux and google chrome version 81.0

time-picker-chrome

**Comment by Betree** [comment]
@snitin315 I tested on Linux/Firefox

File: components/DateTime.js [hunk 2]
```diff
@@ -12,9 +13,9 @@ class DateTime extends React.Component {
   render() {
     const props = this.props;
     const { date, timezone } = props;
-    const value = momentTimezone.tz(new Date(date), timezone);
+    const value = dayjs(new Date(date)).tz(timezone);
```
Comment (snitin315) [inline components/DateTime.js:16]: Yes, it's flaky and I think the issue is with `dayjs` `timezone` plugin. It works fine without it. 
I'm not sure how this should be fixed.

**Comment by snitin315** [comment]
The behavior was flaky but somehow using `toDate()` instead of `$d` fixed it.

File: components/CreateEvent.js [hunk 3]
```diff
@@ -26,7 +25,15 @@ class CreateEvent extends React.Component {
...
-    const timezone = momentTimezone.tz.guess();
+
+    let timezone;
+    // fallback for old browsers
+    if (!Intl.DateTimeFormat || !Intl.DateTimeFormat().resolvedOptions) {
+      timezone = 'utc';
+    } else {
+      timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+    }
+
```
Comment (Betree) [inline components/CreateEvent.js:36]: Can't we use `tz.guess` like we do in `HeroEventDetails.js`?

**Review by Betree — COMMENTED** [review]
Looks good! Let's check this last comment & we'll be good to merge

File: components/CreateEvent.js (hunk 3 above)
Comment (snitin315) [reply components/CreateEvent.js]: Yes we can 🚀

**Pull request merged**